blog/sage-developing-a-smart-contract-based-supply-chain-management-system-on-celo-blockchain-using-python/sage-developing-a-smart-contract-based-supply-chain-management-system-on-celo-blockchain-using-python.md

```diff
@@ -0,0 +1,376 @@
+}
+signed_txn = w3.eth.account.sign_transaction(transaction, private_key)
+transaction_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)
+transaction_receipt = w3.eth.wait_for_transaction_receipt(transaction_hash)
+
+print(f"Product location updated: {transaction_receipt}")
+```
+
+This script shows how to use Web3.py to communicate with a supply chain management smart contract that has been put on the Celo blockchain. It carries out the following duties:
+
+1. Establishes the web3 connection to a Celo node and imports the required libraries.
+2. Introduces the web3.py instance with the geth Proof of Authority (PoA) middleware.
+3. Establishes a contract instance using the ABI and contract address from the deployment script.
+4. Registers a new product using the smart contract's'registerProduct' function, giving it a unique ID, name, and location.
+5. Calls the `getProduct` and `getProductHistory` services of the smart contract to retrieve the product information and history.
+6. Calls the smart contract `transferProductOwnership` function to transfer ownership of the product to a new owner.
+7. Calls the smart contract's `updateProductLocation` method to change the product's location.
+
+Transaction receipts and product details are printed to the console at various points in the script, letting the user track how each operation is going.
+
+Run the following command on your terminal to see the output of your script:
+
+```bash
+python client.py
+```
+
+![Screenshot](https://user-images.githubusercontent.com/104994589/236326562-d3c3102c-a671-4e78-a51c-64d15e732efb.png)
+
+## Conclusion
+
+This tutorial showed us how to use smart contracts on the Celo blockchain to build a supply chain management system. We have looked into the use of smart contracts to increase supply chain efficiency, prevent counterfeit goods, and secure product traceability. We have created a safe and open supply chain system by utilizing the Celo blockchain and Python frameworks like Web3.py.
+
+## Next Steps
+
+Your supply chain management system can be improved still more by:
+
+- Creating an intuitive [front-end application](https://docs.celo.org/blog/tutorials/designing-a-user-friendly-celo-dapp-a-beginners-guide-to-uiux) to facilitate interaction with smart contracts.
+- Extending the [system to accommodate a wider range of supply chain actors](https://consensys.net/blockchain-use-cases/supply-chain-management/), including producers, distributors, merchants, and final customers.
+
+## About the Author
+
+[Israel Okunaya](https://meetisraelokunaya.curious.page/) is an ace writer with a flair for simplifying complexities and a knack for storytelling. He leverages over four years of experience to meet the most demanding writing needs in different niches, especially food and travel, blockchain, and marketing. He sees blockchain as a fascinating yet tricky affair. So, he is given to simplifying its complexities with text and video tutorials.
```

Comment:
can you add your linkedIn and Github profile link somewhere in this section

Reply:
The links are already contained on my portfolio page.